yt/frontends/stream/data_structures.py

```diff
@@ -1069,6 +1072,49 @@ def _setup_data_io(self):
         else:
             self.io = io_registry[self.dataset_type](self.ds)
 
```

Comment:
@jzuhone May I get your review of the following method? It's modified from `StreamHierarchy.update_data` which I believe is written by you. There are several method calls I feel are specific to grid datasets. I've tried my best to check, but I'm not entirely sure. It would be very helpful if someone more familiar with the codebase could take a look.